thesslstore_plugin.php

```diff
@@ -198,10 +198,18 @@ private function order_synchronization(){
 
         $order_query_resp = $api->order_query($order_query_request);
 
-        $services = $this->Services->getList();
+       $sub_query_sql = "SELECT `service_id` FROM `service_fields` WHERE `key` = 'thesslstore_fqdn' GROUP BY `service_id`";
+
+       $services = $this->Record->select()->from("services")
+           ->leftJoin("service_fields", "services.id", "=", "service_fields.service_id", false)
+           ->where("service_fields.key", "=", "thesslstore_order_id")
+           ->where("services.status", "=", "active")
+           ->where("services.id", "notin", array($sub_query_sql), false)
+           ->fetchAll();
```

Comment:
This looks like it can load a lot of data into memory, so it'd be better to fetch and iterate over the PDOStatement object. You can clone the Record object to query with another instance, e.g.:

```
$record = clone $this->Record;
$services = $record->select(['id', 'date_renews'])->from('services')->...->getStatement();

foreach ($services as $service) {
    ...
}
unset($record);
```

Reply:
updated in latest commits